### PR TITLE
Adding an "Include For Ray Tracing" toggle on lights

### PR DESCRIPTION
**Pull request by anisunity**
Adding an "Include For Ray Tracing" toggle on lights to allow the usr to exclude them when ray tracing is enabled in the frame settings of a camera. The option can be found in the advanced mode of the emissive section of the light component. The light is excluded in all the light structures (Tile, Cluster and RT Cluster).

image

Testing status
Ran the DXR tests locally, they are green.
Added ray tracing excluded light to the tests (that tests ray tracing and rasterization).

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs [hunk 1]
```diff
@@ -976,6 +976,8 @@ static void DrawEmissionAdvancedContent(SerializedHDLight serialized, Editor own
                 --EditorGUI.indentLevel;
             }
 
+            EditorGUILayout.PropertyField(serialized.includeForRayTracing, s_Styles.includeLightForRayTracing);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs:979]: I think we should show this field only when Raytracing is enabled. What do you think?

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.Skin.cs [hunk 2]
```diff
@@ -44,6 +44,7 @@ sealed class Styles
             public readonly string cookieNonPOT = "HDRP does not support non power of two cookie textures.";
             public readonly string cookieTooSmall = "Min texture size for cookies is 2x2 pixels.";
             public readonly string cookieBaking = "Light Baking for cookies disabled on the Project Settings.";
+            public readonly GUIContent includeLightForRayTracing = new GUIContent("Include For RayTracing", "When checked, the light will affect the scene for a camera that has the Ray-Tracing frame setting enabled.");
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.Skin.cs:47]: >"Include For RayTracing"
@JordanL8  We are trying to find a label to say that a light is enabled when performing a rendering with a Camera that have a Framesettings "Raytracing" enable.  What do you think is the best wording for this? is Influce for Raytracing good? thanks

Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.Skin.cs:47]: this was discussed with lewis before the PR

Comment (JordanL8) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.Skin.cs:47]: Yeah I thought Include For Raytracing made sense. For the tooltip, how about:

When enabled, the light affects the scene for cameras that have the Ray-Tracing frame setting enabled.

**Review by JordanL8 — CHANGES_REQUESTED** [review]
Left feedback for the tooltip. Also, can you add this property to the [light documentation](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.high-definition/Documentation~/Light-Component.md)?

The description can be:
Indicates whether this light affects the scene for cameras with the **Ray-Tracing** frame setting enabled.

**Review by JulienIgnace-Unity — APPROVED** [review]

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**